lib/main.js

```
@@ -47,108 +42,215 @@ module.exports.runSass = function (options, describe, it) {
-      if (!curTest) {
-        curTest = {test: selInfo.test, assertions: []};
-        curModule.tests.push(curTest);
-      }
-      if (selInfo.output) {
-        curAssertion = _.findWhere(
-          curTest.assertions, {description: selInfo.assert});
-        if (!curAssertion) {
-          curAssertion = {
-            description: selInfo.assert,
-            assert: 'equal',
-          };
-          curTest.assertions.push(curAssertion);
-        }
-        declString = _.map(declPairs, function (pair) {
-          return pair[0] + ': ' + pair[1] + ';';
-        }).join(' ');
-        if (selInfo.type === 'input') {
-          curAssertion.returned = declString;
-        } else { // if (selInfo.type == 'expect') {
-          curAssertion.expected = declString;
-        }
-        if (curAssertion.returned && curAssertion.expected) {
-          curAssertion.passed = curAssertion.returned === curAssertion.expected;
-        }
-      } else {
-        curAssertion = {
-          assert: selInfo.assert,
-          description: (declInfo['-description'] || '""').slice(1, -1),
-          passed: declInfo['-result'] === 'PASS',
-        };
-        _.each(_.pairs(declInfo), function (pair) {
-          if (_.contains(['-expected--', '-returned--'], pair[0].slice(0, 11))) {
-            var attr = pair[0].slice(1, 9);
-            var type = pair[0].slice(11);
-            var val = pair[1];
-            curAssertion[attr] = type + ': ' + val;
-          }
-        });
-        curTest.assertions.push(curAssertion);
-      }
-    });
+
+var parse = module.exports.parse = function (rawCss) {
+  var ast = css.parse(rawCss);
+  var ctx = { modules: [] };
+  var handler = parseModule;
+
+  _.each(ast.stylesheet.rules, function (rule) {
+    handler = handler(rule, ctx);
   });
 
-  return modules;
+  finishCurrentModule(ctx);
+
+  return ctx.modules;
+};
+
+
+var parseError = function (msg, seeking, pos) {
+  throw new Error(
```

I think I would rather return an error here, and throw it when called.

```js
var parseError = function (msg, seeking, pos) {
  return new Error(
    'Line ' + pos.start.line + ', column ' + pos.start.column + ': ' +
       msg + '; looking for ' + seeking);
};
```

```js
throw parseError('Unexpected comment "' + text + '"', 'module header', rule.position);
```